File: src/main/java/j/chain_of_resposibility/inheritance/SupportFactory.java
package j.chain_of_resposibility.inheritance;

public final class SupportFactory {
  public static Support createNoSupport(String name, Support next) {
    return new NoSupport(name, next);
  }

  public static Support createLimitSupport(String name, int limit, Support next) {
    return new LimitSupport(name, limit, next);
  }

  public static Support createLimitSupport(String name, int limit) {
    return new LimitSupport(name, limit);
  }

  public static Support createSpecialSupport(String name, int number, Support next) {
    return new SpecialSupport(name, number, next);
  }

  public static Support createOddSupport(String name, Support next) {
    return new OddSupport(name, next);
  }
}

File: src/main/scala/s/chain_of_resposibility/inheritance/SupportFactory.scala
package s.chain_of_resposibility.inheritance

object SupportFactory {
  def createNoSupport(name: String, next: Support) = new NoSupport(name, next)

  def createLimitSupport(name: String, limit: Int, next: Support) =
    new LimitSupport(name, limit, next)

  def createLimitSupport(name: String, limit: Int) =
    new LimitSupport(name, limit)

  def createSpecialSupport(name: String, number: Int, next: Support) =
    new SpecialSupport(name, number, next)

  def createOddSupport(name: String, next: Support) = new OddSupport(name, next)
}
